server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/AbstractValueAggregatorTests.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.aggregators;
+
+import org.opensearch.index.compositeindex.datacube.MetricStat;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype.StarTreeNumericType;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+public abstract class AbstractValueAggregatorTests extends OpenSearchTestCase {
+
+    private ValueAggregator aggregator;
+
+    @Before
+    public void setup() {
+        aggregator = getValueAggregator();
+    }
+
+    public abstract ValueAggregator getValueAggregator();
+
+    public abstract MetricStat getMetricStat();
+
+    public abstract StarTreeNumericType getValueAggregatorType();
+
+    public void testGetAggregationType() {
+        assertEquals(getMetricStat().getTypeName(), aggregator.getAggregationType().getTypeName());
+    }
+
+    public void testGetAggregatedValueType() {
+        assertEquals(getValueAggregatorType(), aggregator.getAggregatedValueType());
+    }
+
+    public void testGetInitialAggregatedValueForSegmentDocNullValue() {
```

Comment:
Lets add more complex tests which can work with a loop, picking aggregated/segment values in each iteration.

Reply:
such tests are added in ```AbstractStarTreeBuilderTests``` where we sort and reduce documents for all the aggregations

Reply:
I meant unit tests for just aggregator. The main purpose there is testing the star tree, and that tests a broader logic.

Reply:
those tests also ensure that we aggregate properly. I have added new tests as well for the aggregators.